app/code/Magento/MediaStorage/Model/File/Validator/NotProtectedExtension.php

```diff
@@ -78,6 +78,9 @@ protected function _initProtectedFileExtensions()
     {
         if (!$this->_protectedFileExtensions) {
             $extensions = $this->getProtectedFileExtensions();
```

Comment:
The function called here is public and can not be changed without braking parts of Magento. We assume that when nill/null is returned we going to use an empty array as muted value. Ps: With Magento 3 it's better to learn programming without null checks. Thus the getProtectedFileExtensions should have returned ALWAYS an array. That list can then be empty as "null".